### ISSUE TITLE
Content-Type specified (plain/text) must be 'application/json'

### ISSUE THREAD

**Issue by dvohra** [openshift/origin#7873]
Docker version 1.10.2, build c3959b1

The following command generates error `Content-Type specified (plain/text) must be 'application/json'`

`oc rsh <pod>`

![openshift-rsh-pod](https://cloud.githubusercontent.com/assets/3654431/13618586/cd017394-e539-11e5-870e-75d679bf9165.gif)

Output from `oc version`
```
oc v1.1
kubernetes v1.1.0-origin-1107-g4c8e6f4
```

**Comment by jawnsy** [comment]
Hmm, I tried reproducing this problem with the latest build of Origin (built from source) and couldn't do so. Is the error message specific to a particular image?

For me it seems to just work correctly, so maybe the problem was fixed? Or potentially it could be an issue with an intermediate proxy?

```
16:44 fedora@jawnsy-dev:~/go/src/github.com/openshift/origin/assets/app/styles $ oc get pods
NAME                      READY     STATUS              RESTARTS   AGE
etcd-1-166gk              0/1       ContainerCreating   0          1m
etcd-1-deploy             1/1       Running             0          3m
etcd-discovery-1-4djyx    0/1       ContainerCreating   0          1m
etcd-discovery-1-deploy   1/1       Running             0          3m
16:45 fedora@jawnsy-dev:~/go/src/github.com/openshift/origin/assets/app/styles $ oc rsh etcd-1-166gk
sh-4.2$ exit
```

**Comment by dvohra** [comment]
The issue was found with the following image, did not test with another image. 

https://hub.docker.com/r/sath89/oracle-xe-11g/

Does the sath89/oracle-xe-11g image generate the issue?

**Comment by jawnsy** [comment]
Google turns up this issue: https://github.com/kubernetes/kubernetes/issues/20884 which references this https://github.com/kubernetes/kubernetes/issues/19720 which has similar error messages listed.

I'm still using Docker 1.9.1, so that might be why. Can you try downgrading your Docker version and seeing if that helps?

**Comment by smarterclayton** [comment]
It was not supported but we'll put patches in for 1.1.4 and 1.2.
Upstream Kube has not certified on 1.10 yet though so it's likely we
won't officially support it for a while.

**Comment by liggitt** [comment]
Looks like the docker client patched for this. Would need to see what additional changes bumping that would pull in

**Comment by jimmidyson** [comment]
Docker 1.10.3 released with a fix for this issue - docker/docker#20706

**Comment by dvohra** [comment]
Thanks for fixing the issue.

**Comment by cpwais** [comment]
will there be a k8s `1.1.4` patch soon?  this is a super painful bug

**Comment by bladedoyle** [comment]
I still see this issue with Docker 1.10.3.
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"2+", GitVersion:"v1.2.0-alpha.7.874+a7818ac3a1d64a", GitCommit:"a7818ac3a1d64a9a67e0a44e0c07677587bed27f", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"1+", GitVersion:"v1.1.2-dirty", GitCommit:"3085895b8a70a3d985e9320a098e74f545546171", GitTreeState:"dirty"}

$ docker version
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.4.3
 Git commit:   20f81dd-unsupported
 Built:        Sat Apr  2 01:36:24 2016
 OS/Arch:      linux/arm

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.4.3
 Git commit:   20f81dd-unsupported
 Built:        Sat Apr  2 01:36:24 2016
 OS/Arch:      linux/arm

$ kubectl exec -it kibana-gopwz /bin/bash
Content-Type specified (plain/text) must be 'application/json'

[HANG]

**Comment by dvohra** [comment]
Seems like the issue is not fixed in Docker 1.10.3 .

**Comment by dvohra** [comment]
Getting the same issue with

kubernetes v1.1.0-origin-1107-g4c8e6f4
Docker version 1.11.1, build 5604cbe
Ubuntu 3.13.0-85-generic

**Comment by liggitt** [comment]
Fixed in the 1.2 stream in https://github.com/openshift/origin/pull/7269

Kube did not patch in the 1.1 stream
